### PR TITLE
Split cypress pipeline

### LINKED ISSUE #389: Split up CI Pipeline
## Description

The current pipeline takes about +10 minutes to run. It would benefit from some parallelization. The majority of the run time is taken up by the cypress E2E testing step.

## Possible Implementation

Since the jest test suite and the PHP unit test suites need  to be run together (to post reports to code climate), the only other step would be to separate out the Cypress step into its own pipeline. Also add proper caching to the front-end `node_modules` + build assets

### PR DESCRIPTION
**Pull request by willyyhuang**
Closes #389 
Split cypress pipeline due to long sequential pipeline build time

### CODE REVIEW

File: .github/workflows/test-suite.yml [hunk 1]
```diff
@@ -71,6 +71,37 @@ jobs:
             tests/coverage.xml:clover
             coverage/clover.xml:clover
 
+  Cypress:
+    runs-on: ubuntu-latest
+    services:
```
Comment (willyyhuang) [inline .github/workflows/test-suite.yml:76]: i'm not sure if this is needed for `cypress` tbh @m-triassi

Comment (m-triassi) [reply .github/workflows/test-suite.yml:76]: if you mean the `services` block, then thats definitely required, the tests need to be able to read and write data to a database. if you mean the `runs-on` line, that's also required, we need to define what OS to run the pipeline on

Comment (willyyhuang) [reply .github/workflows/test-suite.yml:76]: yeah i was just wondering if the service block was needed

**Review by pascwhale — APPROVED** [review]
Everything looks good, it seems like you copied the right steps for the new pipeline. I'm happy that this will lessen the delay of our checks.

**Review by m-triassi — APPROVED** [review]
Agree'd this is a good stop gap until we find a better way to speed up the cypress suite in general

File: .github/workflows/test-suite.yml (hunk 1 above)
Comment (m-triassi) [reply .github/workflows/test-suite.yml:82]: i see, yeah services is definitely needed

**Pull request merged**